src/js/actions/layers.js

```diff
@@ -134,6 +132,135 @@ define(function (require, exports) {
+        return {
+            name: "get",
+            descriptor: {
+                "null": {
+                    "_multiGetRef": [
+                        {
+                            _propertyList: properties
+                        },
+                        layerRef,
+                        docRef
+                    ]
+                }
+                
+            },
+            "options": {
+                useMultiGet: true,
+                failOnMissingProperty: false
+            }
+        };
+    };
+    
+    var getLayerDescriptors = function (document, layers, options) {
+        layers = layers.toArray();
+        
+        if (layers.length === 0) {
+            return Promise.resolve([]);
+        }
+
+        options = _.merge({
+            getEssentialProperties: true,
+            getBoundsProperties: true,
+            getSelectedProperties: true,
+            getExtensionData: true
+        }, options);
+
+        var docRef = documentLib.referenceBy.id(document.id),
+            essentialDescriptorCommands = [],
+            selectedDescriptorCommands = [];
+            
+        layers.forEach(function (layer) {
+            var layerRef = layerLib.referenceBy.id(layer.id);
+            
+            if (options.getEssentialProperties) {
+                essentialDescriptorCommands.push(multiGetRef(docRef, layerRef, _layerProperties));
+            }
+            if (options.getSelectedProperties) {
+                selectedDescriptorCommands.push(multiGetRef(docRef, layerRef, _lazySelectedLayerProperties));
+            }
+        });
+        
+        var essentialPropertiesPromise = essentialDescriptorCommands.length === 0 ? Promise.resolve([]) :
+                descriptor.batchPlay(essentialDescriptorCommands),
+            selectedPropertiesPromise = selectedDescriptorCommands.length === 0 ? Promise.resolve([]) :
+                descriptor.batchPlay(selectedDescriptorCommands);
+
+        var extensionPromise = Promise.resolve([]);
+        if (options.getExtensionData) {
+            var extensionPlayObjects = layers.map(function (layer) {
+                var layerRef = layerLib.referenceBy.id(layer.id);
+
+                return layerLib.getExtensionData(docRef, layerRef, METADATA_NAMESPACE);
+            });
+
+            extensionPromise = descriptor.batchPlayObjects(extensionPlayObjects)
+                .map(function (extensionData) {
+                    var extensionDataRoot = extensionData[METADATA_NAMESPACE];
+                    return (extensionDataRoot && extensionDataRoot.exportsMetadata) || {};
+                });
+        }
+        
+
+        return Promise.join(
+            essentialPropertiesPromise,
+            selectedPropertiesPromise,
+            extensionPromise,
+            function (essential, selected, extension) {
+                selected.forEach(function (properties) {
+                    properties.initialized = true;
```

Comment:
Mark the `descriptor.initialized` so that the Layer model knows it is fully initialized.